### PR TITLE
[DPR2-189] Use time instead of date field in ext. moves.

### PR DESCRIPTION
**Pull request by stalwartsolutions**
Also add date-time boundary filtering test, and update library version for filtering tweak.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ConfiguredApiIntegrationTest.kt [hunk 1]
```diff
@@ -305,7 +305,7 @@ class ConfiguredApiIntegrationTest : IntegrationTestBase() {
       4,
       7849,
       LocalDateTime.of(2023, 5, 1, 0, 0, 0),
-      LocalDateTime.of(2023, 5, 1, 15, 19, 0),
+      LocalDateTime.of(2023, 5, 20, 15, 19, 0),
```
[the remark this answers is not in the document]
Comment (gavriil-g-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ConfiguredApiIntegrationTest.kt:308]: Or is it needed here?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ConfiguredApiIntegrationTest.kt [hunk 2]
```diff
@@ -373,7 +373,7 @@ class ConfiguredApiIntegrationTest : IntegrationTestBase() {
 
     val movementPrisoner3 = mapOf(PRISON_NUMBER to "G3418VR", NAME to "LastName3, F", DATE to "2023-04-30", DIRECTION to "In", TYPE to "Transfer", ORIGIN to "BEDFORD (HMP)", ORIGIN_CODE to "BFI", DESTINATION to "NORTH SEA CAMP (HMP)", DESTINATION_CODE to "NSI", REASON to "Transfer In from Other Establishment")
 
-    val movementPrisoner4 = mapOf(PRISON_NUMBER to "G3411VR", NAME to "LastName5, F", DATE to "2023-05-01", DIRECTION to "Out", TYPE to "Transfer", ORIGIN to "Lowestoft (North East Suffolk) Magistrat", ORIGIN_CODE to "LWSTMC", DESTINATION to "WANDSWORTH (HMP)", DESTINATION_CODE to "WWI", REASON to "Transfer Out to Other Establishment")
+    val movementPrisoner4 = mapOf(PRISON_NUMBER to "G3411VR", NAME to "LastName5, F", DATE to "2023-05-20", DIRECTION to "Out", TYPE to "Transfer", ORIGIN to "Lowestoft (North East Suffolk) Magistrat", ORIGIN_CODE to "LWSTMC", DESTINATION to "WANDSWORTH (HMP)", DESTINATION_CODE to "WWI", REASON to "Transfer Out to Other Establishment")
```
Comment (gavriil-g-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ConfiguredApiIntegrationTest.kt:376]: Also is this date change needed?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ConfiguredApiIntegrationTest.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (stalwartsolutions) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ConfiguredApiIntegrationTest.kt:308]: It's needed to bring it to the boundary in this case.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ConfiguredApiIntegrationTest.kt (hunk 2 above)
Comment (stalwartsolutions) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ConfiguredApiIntegrationTest.kt:376]: Yes, it's the counterpart to the about change to bring the movement to the query boundary.

**Review by gavriil-g-moj — APPROVED** [review]
LGTM

**Pull request merged**